cli/app.go

```diff
@@ -61,16 +65,56 @@ Examples:
 `)
 }
 
+func promptYesNo(msg string) (result bool) {
+	fmt.Print(msg)
+	fmt.Print(" (yes/no): ")
+	for {
+		var answer string
+		fmt.Scanln(&answer)
+		switch answer {
+		case "y", "yes":
+			return true
+		case "n", "no":
+			return false
+		default:
+			fmt.Print("Please type 'yes' or 'no': ")
+		}
+	}
+}
+
 func runCreate(args *docopt.Args, client *controller.Client) error {
 	app := &ct.App{}
 	app.Name = args.String["<name>"]
+	remote := args.String["<remote>"]
 
+	// Test if remote name exists and prompt user
+	remotes, err := gitRemoteNames()
+	if err != nil {
+		return err
+	}
+
+	if !args.Bool["--yes"] {
+		for _, r := range remotes {
+			if r == remote {
+				fmt.Println("There is already a git remote called", remote)
+				if !promptYesNo("Are you sure you want to replace it?") {
+					log.Println("The app was not created. Please, declare the desired local git remote name with --remote flag.")
+					return nil
+				}
+			}
+		}
+	}
+
+	// Create the app
 	if err := client.CreateApp(app); err != nil {
 		return err
 	}
 
-	exec.Command("git", "remote", "remove", "flynn").Run()
-	exec.Command("git", "remote", "add", "flynn", gitURLPre(clusterConf.GitHost)+app.Name+gitURLSuf).Run()
+	// Register git remote
+	if remote != "" {
+		exec.Command("git", "remote", "remove", remote).Run()
+		exec.Command("git", "remote", "add", remote, gitURLPre(clusterConf.GitHost)+app.Name+gitURLSuf).Run()
```

Comment:
We should add an "--" arg before the final two chunks in this `git remote add.  Since the add subcommand takes a variety of other arguments, this will make it absolutely clear to git that these are meant to be the `<name> <url>` chunks.

In this case, the `gitURLPre` function already massages input such that it covers all injections I can currently think of, so the prior state doesn't seem like an active security/stability hole, but if the `remote` value is user input, we definitely want this protection now (and I'd advocate doing it just out of habit for clarity and defense in depth regardless; but now's as good a time as any :) ).

(The adjacent `git remote remove` command supports no other optional arguments, so it doesn't require an end-of-options signifier... looks like it doesn't accept one either.)